xwayland/xwm.c

```diff
@@ -935,15 +936,25 @@ static void xwm_handle_configure_notify(struct wlr_xwm *xwm,
 		return;
 	}
 
-	xsurface->x = ev->x;
-	xsurface->y = ev->y;
-	xsurface->width = ev->width;
-	xsurface->height = ev->height;
+	bool geometry_changed =
+		(xsurface->x != ev->x || xsurface->y != ev->y ||
+		 xsurface->width != ev->width || xsurface->height || ev->height);
```

Comment:
Typo: `xsurface->height || ev->height` should read `xsurface->height != ev->height`

Reply:
very nice catch!